applications/StructuralMechanicsApplication/custom_conditions/line_load_condition.cpp

```diff
@@ -0,0 +1,456 @@
+        // Declaring tangent and Jacobian
+        array_1d<double, 3> tangent_xi, tangent_eta;
+        Matrix J(TDim, 1);
+
+        // Getting LOCAL_AXIS_2
+        GetLocalAxis2(tangent_eta);
+
+        // Iterate over the Gauss points
+        for (IndexType point_number = 0; point_number < r_integration_points.size(); ++point_number) {
+            r_geometry.Jacobian(J, point_number, integration_method);
+
+            // Definition of the tangent
+            GetLocalAxis1(tangent_xi, J);
+
+            // Computing normal
+            MathUtils<double>::UnitCrossProduct(rOutput[point_number], tangent_xi, tangent_eta);
+        }
+    } else {
+        for (IndexType point_number = 0; point_number < r_integration_points.size(); ++point_number) {
+            rOutput[point_number] = ZeroVector(3);
+        }
+    }
+
+    KRATOS_CATCH( "" );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void LineLoadCondition<TDim>::CalculateAll(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo,
+    const bool CalculateStiffnessMatrixFlag,
+    const bool CalculateResidualVectorFlag
+    )
+{
+    KRATOS_TRY;
+
+    const auto& r_geometry = GetGeometry();
+    const SizeType number_of_nodes = r_geometry.size();
+    const SizeType block_size = this->GetBlockSize();
+
+    // Resizing as needed the LHS
+    const SizeType mat_size = number_of_nodes * block_size;
+
+    if ( CalculateStiffnessMatrixFlag ) { // Calculation of the matrix is required
+        if ( rLeftHandSideMatrix.size1() != mat_size ) {
+            rLeftHandSideMatrix.resize( mat_size, mat_size, false );
+        }
+        noalias( rLeftHandSideMatrix ) = ZeroMatrix( mat_size, mat_size ); //resetting LHS
+    }
+
+    // Resizing as needed the RHS
+    if ( CalculateResidualVectorFlag ) { // Calculation of the matrix is required
+        if ( rRightHandSideVector.size( ) != mat_size ) {
+            rRightHandSideVector.resize( mat_size, false );
+        }
+        noalias( rRightHandSideVector ) = ZeroVector( mat_size ); //resetting RHS
+    }
+
+
+    // Reading integration points and local gradients
+    const IntegrationMethod integration_method = IntegrationUtilities::GetIntegrationMethodForExactMassMatrixEvaluation(r_geometry);
+    const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints(integration_method);
+    const GeometryType::ShapeFunctionsGradientsType& rDN_De = r_geometry.ShapeFunctionsLocalGradients(integration_method);
+    const Matrix& rNcontainer = r_geometry.ShapeFunctionsValues(integration_method);
+
+    // Sizing work matrices
+    Vector pressure_on_nodes = ZeroVector( number_of_nodes );
+
+    // Pressure applied to the element itself
+    double pressure_on_condition = 0.0;
```

Comment:
what we discussed yesterday is that if it is a 3d line it makes no sense to apply a follower pressure (of any type).
Pressures should be dealt with by surfaces in the 3D case.

the proposal would hence be to simply ignore the pressure when you are in the 3D case. you can use instead a LINE_LOAD which is a vector and intrinsically contains the direction

Reply:
HI, yes for sure I do not want to apply a pressure, since it requires surfaces. But a distributed load on a line would be perfect. 
On structural 3D, using Gid interface, the LINELOAD is not available yet.

Reply:
First we add to the master on Kratos, later @jginternational will add it

Reply:
In fact, anyone can add it! Come on, there is no TCL, it's just some xml copy/paste! 

![tenor](https://user-images.githubusercontent.com/5918085/59427649-744bf500-8ddb-11e9-8767-977b7ab6e457.gif)

Reply:
If it is so easy, why are avoiding it?, it is something it takes for you 4 seconds, and for us 2 hours

Reply:
Thats exactly why I proposed #4773

Reply:
@philbucher Can you approve??

Reply:
Will do once I am back in my office, i.e. later this afternoon

Reply:
@loumalouomega we did the GUI data in xml just for this case. A developer creates a new element or load, and it's easy to add it. maybe it's 4 seconds for me but I'm sure that's 2 minutes for a newbie. 
As U2 says, "if you never try, you'll never know"

Reply:
Can you understand that I have other things to do?

Reply:
In this case I agree with @jginternational . It shouldn't be too hard to add a new condition to the existent ones (copy & paste one above and change name, Kratos name and not much more). 

Furthermore, we should take into account that we are around 80 developers and just 1 part-time GUI developer. Having this in mind, we must collaborate with him as much as possible (for the sake of the common benefit) if we want the GUI to advance as fast as the project does . Thus, I think that if someone wants such kind of minor features to be available in his/her GUI should at least try. I'm convinced that if he/she doesn't succeed, @jginternational will support him.

BTW @loumalouomega , no one explicitly asked you to do it. No worries, we all know that you are involved in (probably too) many things. :wink: